### ISSUE TITLE
dask.distributed fails with too many jobs but works fine with a few.

### ISSUE THREAD

**Issue by naji-s** [dask/distributed#3089]
I am trying to run multiple ten/hundred copies of a few machine learning models each with their cross-validation using dask. This fails with the following log file, when the number of parallel runs is larger (this number is controlled with ```num_of_bootstrap_trials``` as described below, after the code). 

```
 Traceback (most recent call last):
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/distributed/cli/dask_worker.py", line 416, in <module>
    go()
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/distributed/cli/dask_worker.py", line 412, in go
    main()
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/nshajari/anaconda3/envs/lepu_py/lib/python3.7/site-packages/distributed/cli/dask_worker.py", line 403, in main
    raise TimeoutError("Timed out starting worker.") from None
tornado.util.TimeoutError: Timed out starting worker.
```

The code is as follows:
```
if __name__ == '__main__':  
    __spec__ = None

    import dask
    from MyModel import MyModel
    import sys, os
    from time import time
    from dask import compute, delayed, config
    import dill as pickle
    from sklearn.model_selection import train_test_split, cross_val_score
    from psych_model import psych_model
    from sklearn.model_selection import KFold
    from pathos.multiprocessing import ProcessingPool
    import numpy as np
    from numpy import random, isnan, genfromtxt, hstack, asarray, logspace, zeros
    from naive_clf import NaiveLePU
    from sklearn import metrics
    import dill as pickle
    from sklearn.utils import resample
    from sklearn.externals.joblib import Parallel, parallel_backend, register_parallel_backend


    def report_results(model_dict):
        """ THE FUNCTION THAT DOES THE LEARNING WITH CROSS-VALIDATION AND RUNS IN PARALLEL"""
        from sklearn.pipeline import Pipeline
        from sklearn import metrics
        from MyModel import MyModel
        from numpy import genfromtxt
        import numpy as np
        from numpy import random, isnan
        import dill as pickle
        from numpy import random, hstack, asarray, logspace, zeros
        from sklearn.utils import resample
        from sklearn.externals.joblib import Parallel, parallel_backend, register_parallel_backend
        from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
        from sklearn.model_selection import StratifiedKFold
        from dask_ml.model_selection import GridSearchCV

        X = model_dict['X']
        y = model_dict['y']
        l = model_dict['l']


        param_grid = model_dict['param_grid']
        clf = model_dict['clf']
        model_name = model_dict['model_name']
        score_func = model_dict['score_func']

        train_test_ratio = 0.1
        X_train, X_test, \
        y_train, y_test, \
        l_train, l_test= train_test_split(X, y, l, test_size=train_test_ratio, stratify=y)


        pipeline = Pipeline([
            ('vect', CountVectorizer()),
            ('tfidf', TfidfTransformer()),
            (model_name, clf),
        ])
        strat_kfold = StratifiedKFold(4)#, random_state=np.random.get_state())

        model_cv = GridSearchCV(
            pipeline,
            scoring=score_func,
            cv=strat_kfold,
            param_grid=param_grid
        )
        model_cv.fit(X_train, y_train)
eg_coeff'] = model_cv.best_estimator_.psych_reg_coeff
        return model_cv


    
    
    # subject number
    subject = 1
    i = subject_num = int(subject)
    process_id = "TEST" 
    
    pickle_file_location = '/home/*****/tf_idf.pkl'
        output = '/home/*****/output/'
    else:
        raise NotImplementedError

    if not os.path.exists(output + str(subject_num) + '/' + process_id):
           os.makedirs(output + str(subject_num) + '/' + process_id)


    ########################################################################################################################
    ######################## READ DATA
    ################################################################################################

    main_X = np.asarray(data_dict['X_text'])
    main_y = np.asarray(data_dict['y'])
    main_l = np.asarray(data_dict['s_'+str(subject)]) & np.asarray(data_dict['y'])

    
    ########################################################################################################################
    ######################## SETTING UP THE CLUSTER
    ################################################################################################

    from dask_jobqueue import SLURMCluster
    cluster = SLURMCluster(
                           cores=24,
                           processes=4, 
                         memory='60GB',
                         nanny=True,
                        interface='ib0',
                        dashboard_address=':12123',
                        silence_log='info',
                        log_directory='/home/****/master_thesis/dask_logs',
                        walltime='00:03:00')
    cluster.scale(cores=48)  # Start 100 workers in 100 jobs that match the description above
    from dask.distributed import Client
    sleep(10)
    client = Client(cluster, processes=False)

    ########################################################################################################################
    ######################## DEFINING MODELS 
    ################################################################################################
        
    num_of_bootstrap_trials = 5
    from time import sleep


    def custome_scorer(estimator, X, l):
        from numpy import isnan
        from sklearn import metrics
        estimator_outputs = estimator.predict_proba(X)
        if isnan(estimator_outputs).any():
            print(type(estimator).__name__)
        return metrics.roc_auc_score(l, estimator_outputs)


    process_loc = output + str(subject_num) + '/' + process_id + '/'

    def custome_scorer_2(estimator, X, y):
        return -metrics.brier_score_loss(y, estimator.predict_proba(X))



    import dask.dataframe as dd
    import pandas as pd
    

    full_bootstreap_idx_arr = np.random.choice(np.arange(dataset_size), size=(num_of_bootstrap_trials, dataset_size, len(all_model_list)))
    print (full_bootstreap_idx_arr[0, :, 0])
    model_list = []
    
    
    for i in range(num_of_bootstrap_trials):


        if 'naive_lepu' in active_model_list:
            model_dict = dict()
            model_dict['model_name'] = 'naive_lepu'
            model_dict['score_func'] = LePU_scorer
            model_dict['param_grid'] = {'naive_lepu__C': np.logspace(-4, 4, 5)}
            model_dict['clf'] = NaiveLePU()
            model_dict['X'] = main_X[full_bootstreap_idx_arr[i, :, 5]]
            model_dict['l'] = main_l[full_bootstreap_idx_arr[i, :, 5]]
            model_dict['y'] = main_y[full_bootstreap_idx_arr[i, :, 5]]
            model_list.append(model_dict)

        if 'real_clf' in active_model_list:
            model_dict = dict()
            model_dict['model_name'] = 'real_clf'
            model_dict['score_func'] = real_clf_scorer
            model_dict['param_grid'] = {'real_clf__C': np.logspace(-4, 4, 5)}
            model_dict['clf'] = MyModel()
            model_dict['X'] = main_X[full_bootstreap_idx_arr[i, :, 6]]
            model_dict['l'] = main_l[full_bootstreap_idx_arr[i, :, 6]]
            model_dict['y'] = main_y[full_bootstreap_idx_arr[i, :, 6]]
            model_list.append(model_dict)


    parallel_time_start = time()
    with parallel_backend(backend):
        values = [delayed(report_results)(model) for model in model_list]
        model_cv_list = compute(*values)

    print ("The process time was:", time()- parallel_time_start, "seconds...")
```

```report_results``` is the function that runs paralell. In this function there is CrossValidation happening for model selection.  Each report result only handles one model. 

I am not sure why do I get error when I increase ```num_of_bootstrap_trials``` which is basically the number of times ```report_results``` is called!

I would appreciate any help regarding the error which seems to do with ```TimeoutError```?

Thank you!

**Comment by naji-s** [comment]
This was not exactly the bug. I will replicate the bug and post back and open this. Thanks!

**Comment by naji-s** [comment]
Reopenning as now the error is exactly what I get.

**Comment by mrocklin** [comment]
The error seems to be about difficulty starting workers.  Is all of your code about training models necessary to reproduce this error?

https://blog.dask.org/2018/02/28/minimal-bug-reports

**Comment by jrbourbeau** [comment]
Perhaps this is a duplicate of https://github.com/dask/dask-jobqueue/issues/309? @naji-s you might consider running the example in that issue to see if you can reproduce the `TimeoutError`

**Comment by mrocklin** [comment]
Closing due to lack of response / minimal reproducible example.  If you're able to provide a minimal reproducible example then please share it here and we'll reopen this.
